src/components/Card.js

```diff
@@ -1,38 +1,63 @@
 class Card {
-  constructor({ name, link }, cardSelector, handleImageClick) {
+  constructor(
+    { name, link, cardID, isLiked, _id, cardData },
+    cardSelector,
+    handleImageClick,
+    handleLikeClick,
+    handleDeleteSubmit
+  ) {
     this._name = name;
     this._link = link;
+    this._cardID = cardID;
+    this._id = _id;
+    this.isLiked = isLiked;
+    this.cardData = cardData;
     this._cardSelector = cardSelector;
     this._handleImageClick = handleImageClick;
+    this._handleLikeClick = handleLikeClick;
+    this._handleDeleteSubmit = handleDeleteSubmit;
   }
 
   _setEventListeners() {
-    this._cardElement
-      .querySelector(".card__like-button")
-      .addEventListener("click", () => {
-        this._handleLikeButton();
-      });
-
-    this._cardElement
-      .querySelector(".card__delete-button")
-      .addEventListener("click", () => {
-        this._handleDeleteButton();
-      });
+    this._likeButton = this._cardElement.querySelector(".card__like-button");
+
+    this._likeButton.addEventListener("click", () => {
+      this._handleLikeClick(this);
+    });
+
+    this._deleteButton.addEventListener("click", () => {
+      this._handleDeleteSubmit(this._id, this);
+    });
 
     this._cardImage.addEventListener("click", () => {
       this._handleImageClick({ link: this._link, name: this._name });
     });
   }
 
-  _handleDeleteButton() {
+  handleDeleteSubmit() {
+    const cardID = this.cardData;
+    handleDeleteButton(cardID);
+  }
+
+  isLiked() {
+    return this.isLiked;
+  }
+
+  _setLikes() {
+    if (this.isLiked) {
+      this._likeButton.classList.add("card__like-button_active");
+    } else {
+      this._likeButton.classList.remove("card__like-button_active");
+    }
+  }
+
+  removeCard() {
     this._cardElement.remove();
     this._cardElement = null;
   }
 
-  _handleLikeButton() {
-    this._cardElement
-      .querySelector(".card__like-button")
-      .classList.toggle("card__like-button_active");
+  handleLikeButton() {
```

Comment:
Needs correction: there is no need to have such method. Please remove it and create a method for updating the likes instead, which you will be use in index.js after updating the likes.

```js
// call this method from index.js after successful api response in `then` block
updateLikes(isLiked) { 
   // set instance variable
    this._isLiked = isLiked; 
    this._renderLikes(); 
} 
```

Reply:
I updated the code as recommended. Thank you.